File: tests/python/aws-bedrock-agent/test_prototype.py
"""Conformance test: AWS Bedrock Agent invoke_agent with manual instrumentation.

Exercises: invoke_agent (Bedrock Agent Runtime InvokeAgent API)
against a mock Bedrock server, with manual span instrumentation.
"""

from common import run, run_invoke_agent


def instrument():
    """No-op: manual instrumentation only."""
    pass


if __name__ == "__main__":
    run(
        "Manual: AWS Bedrock Agent Invoke Agent Conformance Test",
        instrument,
        [run_invoke_agent],
    )

File: tests/js/aws-bedrock-agent/test_prototype.ts
/**
 * Conformance test: Manual invoke_agent instrumentation for AWS Bedrock Agent.
 *
 * Exercises: invoke_agent (Bedrock Agent Runtime InvokeAgent API)
 * with manual OTel spans.
 */

import { setupOtel, runInvokeAgent } from "./common";
import { flushAndShutdownOtel } from "../otel";

async function main() {
  console.log("=== Manual: AWS Bedrock Agent JS Invoke Agent Conformance Test ===");

  const otel = setupOtel();

  await runInvokeAgent();

  await flushAndShutdownOtel(otel);
}

main().catch((e) => { console.error(e); process.exit(1); });
